scripts/spokepool.ts

```
@@ -255,6 +266,70 @@ async function fetchTxn(args: Record<string, number | string>, _signer: Signer):
   return true;
 }
 
+async function fetchDepositId(args: Record<string, number | string>): Promise<boolean> {
+  const originChainId = Number(args.fromChainId);
+  const depositId = Number(args.depositId);
+  const destinationChain = Number(args.toChainId);
+  const depositor = String(args.depositor);
+  const hours = Number(args.hours) || 24;
+
+  if (!utils.validateChainIds([originChainId]) || Number.isNaN(depositId) || depositId < 0) {
+    usage(); // no return
+  }
+  const provider = new ethers.providers.StaticJsonRpcProvider(utils.getProviderUrl(originChainId));
+  const voidSigner = new VoidSigner(randomAddress(), provider);
+  const {
+    clients: { hubPoolClient },
+    config,
+  } = await createDataworker(Logger, voidSigner);
+  const latestBlockOnChain = await provider.getBlock("latest");
+  const previousBlockOnChain = await provider.getBlock(latestBlockOnChain.number - 1);
+  const secondsDifference = latestBlockOnChain.timestamp - previousBlockOnChain.timestamp;
+  const blocksPerHour = 3600 / secondsDifference;
+  const blockLookback = Math.floor(hours * blocksPerHour);
+
+  const spokePoolClients = await constructSpokePoolClientsWithStartBlocks(
+    Logger,
+    hubPoolClient,
+    config,
+    voidSigner,
+    {
+      [originChainId]: latestBlockOnChain.number - blockLookback,
+    },
+    {
+      [originChainId]: latestBlockOnChain.number,
+    },
+    [originChainId]
+  );
+
+  const spokeClient = spokePoolClients[originChainId];
+
+  await spokeClient.update();
+
+  const deposit = await spokeClient.findDeposit(depositId, destinationChain, depositor);
```

nit: if undefined, should we throw a detailed error recommending caller to set longer `hours`?

---

Good call - done